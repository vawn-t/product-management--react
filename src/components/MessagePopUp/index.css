.popup-message {
  display: none;
  position: fixed;
  border-radius: 10px;
  line-height: 2rem;
  padding: 0 1rem;
  top: 1rem;
  right: 2rem;
}

.popup-error {
  background-color: var(--danger);
}

.popup-success {
  background-color: var(--success);
}

.popup-message.active {
  display: block;
}
