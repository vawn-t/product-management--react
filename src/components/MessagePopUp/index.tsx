// Libraries
import React, { memo, useEffect, useState } from 'react';

// Stores
import { clearMessages, useStore } from '@store/index';

// Components
import { Text } from '@components/index';

// Styles
import './index.css';

export interface MessagePopUpProps {
  successMessage: string;
  errorMessage: string;
  isError: boolean;
}

const MessagePopUp: React.FC<MessagePopUpProps> = ({
  successMessage,
  errorMessage,
  isError
}) => {
  const { dispatch } = useStore();
  const [isVisible, setIsVisible] = useState<boolean>(true);

  useEffect(() => {
    const timer = setTimeout(() => {
      setIsVisible(false);
      dispatch(clearMessages());
    }, 2000);

    return () => clearTimeout(timer);
  }, []);

  return (
    <div
      className={`popup-message${isVisible ? ' active' : ''}${
        isError ? ' popup-error' : ' popup-success'
      }`}
    >
      <Text color='var(--white)'>
        {isError ? errorMessage : successMessage}
      </Text>
    </div>
  );
};

export default memo(MessagePopUp);
