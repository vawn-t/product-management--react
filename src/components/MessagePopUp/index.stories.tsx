// Libraries
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Components
import MessagePopUp from './index';

// Styles
import '@assets/styles/reset.css';
import '@assets/styles/App.css';
import '@assets/styles/variables.css';

export default {
  title: 'Components/MessagePopUp',
  component: MessagePopUp
} as ComponentMeta<typeof MessagePopUp>;

const Template: ComponentStory<typeof MessagePopUp> = (args) => (
  <MessagePopUp {...args} />
);

export const Error = Template.bind({});
Error.args = {
  errorMessage: 'Error Message',
  isError: true
};

export const Success = Template.bind({});
Success.args = {
  successMessage: 'Success Message'
};
