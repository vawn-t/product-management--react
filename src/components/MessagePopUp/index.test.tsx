// Libraries
import React from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';

// Components
import MessagePopUp, { MessagePopUpProps } from './index';

describe('MessagePopUp component', () => {
  const defaultProps: MessagePopUpProps = {
    errorMessage: '',
    isError: false,
    successMessage: 'Success message'
  };

  test('should render correctly', () => {
    const defaultTree = create(<MessagePopUp {...defaultProps} />).toJSON();
    expect(defaultTree).toMatchSnapshot();
  });

  test('wait 2 second before disappear', () => {
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');

    render(<MessagePopUp {...defaultProps} />);

    expect(setTimeout).toHaveBeenCalled();
  });
});
