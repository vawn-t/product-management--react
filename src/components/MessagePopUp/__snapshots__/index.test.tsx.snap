// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagePopUp component should render correctly 1`] = `
<div
  className="popup-message active popup-success"
>
  <p
    className="text-default"
    style={
      Object {
        "color": "var(--white)",
        "fontSize": undefined,
        "fontStyle": "inherit",
        "fontWeight": 400,
      }
    }
  >
    Success message
  </p>
</div>
`;
